### PR TITLE
Introduce gatsby-plugin-clerk v3

### PR DESCRIPTION
**Pull request by nikosdouvlis**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [x] `@clerk/gatsby`
- [ ] `@clerk/types`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

❗ Work in progress

- Move `gatsby-plugin-clerk` into our monorepo
- Support Gatsby SSR using the latest Clerk v3 SDK
- Support Gatsby's `getServerData`

More details about the proposed Gatsby SSR API will follow soon :)

### CODE REVIEW

File: packages/gatsby-plugin-clerk/src/ssr/withServerAuth.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      origin: headers.get('origin'),
+      host: headers.get('host') as string,
+      forwardedPort: headers.get('x-forwarded-port') as string,
+      forwardedHost: headers.get('x-forwarded-host') as string,
+      referrer: headers.get('referer'),
+      userAgent: headers.get('user-agent') as string,
+      fetchInterstitial: () => Promise.resolve(''),
+    });
+
+    console.log('getAuthData', {
```
Comment (SokratisVidros) [inline packages/gatsby-plugin-clerk/src/ssr/withServerAuth.ts:39]: 🙃 Please remove

**Review by npetridis — APPROVED** [review]

**Review by SokratisVidros — APPROVED** [review]

File: packages/gatsby-plugin-clerk/package.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    "dist",
+    "src",
+    "gatsby-browser.js",
+    "gatsby-ssr.js",
+    "gatsby-node.js",
+    "ssr.js",
+    "ssr.d.ts"
+  ],
+  "scripts": {
+    "prepublishOnly": "npm run build",
```
Comment (igneel64) [inline packages/gatsby-plugin-clerk/package.json:29]: 🙃 
We don't need that I think right ?

File: packages/gatsby-plugin-clerk/src/GatsbyClerkProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      ) : (
+        children
+      )}
+    </ReactClerkProvider>
+  );
+}
+
+const Reload = () => {
+  React.useEffect(() => {
+    // TODO: Add fix for firefox edge case
```
Comment (igneel64) [inline packages/gatsby-plugin-clerk/src/GatsbyClerkProvider.tsx:41]: ❓ 
The way to fix this is already in the latest interstitial code, right, can't we use the same ?

File: packages/gatsby-plugin-clerk/src/ssr/types.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+} & (Options extends { loadSession: true } ? { session: Session | null } : {}) &
+  (Options extends { loadUser: true } ? { user: User | null } : {});
+
+export type WithServerAuthCallback<Return, Options> = (props: GetServerDataPropsWithAuth<Options>) => Return;
+
+export type WithServerAuthOptions = {
+  loadUser?: boolean;
+  loadSession?: boolean;
+  // TODO: Add support for the following options
+  // jwtKey?: string;
```
Comment (igneel64) [inline packages/gatsby-plugin-clerk/src/ssr/types.ts:16]: ❓ 
Why is this off ?

File: packages/gatsby-plugin-clerk/src/GatsbyClerkProvider.tsx (hunk 3 above)
Comment (nikosdouvlis) [reply packages/gatsby-plugin-clerk/src/GatsbyClerkProvider.tsx:41]: Thanks for the review @igneel64 ! Yes you're right, its going to be added immediately after the beta is released. The next version will include the FF fix and observability

**Pull request merged**